File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.1 - Controle Condicional Verdadeiro/Iterativo/enquantoIT.swift
import Foundation

var I: Int = 0
var N: Int = 0
var R: Int = 0

print("LACO PRE-TESTE COM FLUXO VERDADEIRO")
print("Multiplicacao de numero por 3 por 5 vezes\n")
I = 1
while I <= 5 {
    print("Calculo: \(I)")
    print("Entre com um valor numerico inteiro: ", terminator: "")
    if let input = readLine(), let numericInput = Int(input) {
        N = numericInput
        R = N * 3
        print("O numero informado X 3 = \(R)\n")
    } else {
        print("Entrada invalida. Por favor, insira um valor numerico inteiro.\n")
    }
    I = I + 1
}

File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.1 - Controle Condicional Verdadeiro/Efetivo/Iterativo/enquantoIT.ts
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 5 - Topico 5.3.1
 * 
 * Programa ..: LACO CONDICIONAL PRE-TESTE COM FLUXO VERDADEIRO ITERATIVO
 */

import readlineSync from 'readline-sync';

console.log('LACO PRE-TESTE COM FLUXO VERDADEIRO');
console.log('Multiplicacao de numero por 3 por 5 vezes');
console.log();
let I: number = 1;
while (I <= 5) {
  console.log('Calculo:', I);
  const N: number = parseInt(readlineSync.question('Entre valor numerico inteiro: '));
  const R: number = N * 3;
  console.log('O numero informado X 3 = ', R);
  console.log();
  I = I + 1;
}
